src/runtime-rs/crates/hypervisor/src/kernel_param.rs

```diff
@@ -67,43 +69,37 @@ impl KernelParams {
         Self { params }
     }
 
-    pub(crate) fn new_rootfs_kernel_params(rootfs_driver: &str) -> Self {
-        let params = match rootfs_driver {
-            VM_ROOTFS_DRIVER_BLK => {
-                vec![
-                    Param {
-                        key: "root".to_string(),
-                        value: "/dev/vda1".to_string(),
-                    },
-                    Param {
-                        key: "rootflags".to_string(),
-                        value: "data=ordered,errors=remount-ro ro".to_string(),
-                    },
-                    Param {
-                        key: "rootfstype".to_string(),
-                        value: "ext4".to_string(),
-                    },
-                ]
+    pub(crate) fn new_rootfs_kernel_params(rootfs_driver: &str, rootfs_type: &str) -> Result<Self> {
```

Comment:
Thinking long-term here, have you considered how to achieve that result when you cannot easily change the command-line parameters (e.g. in the confidential containers case)?

Reply:
Personally I'd suggest it as an unblocked stuff here but yeah confidential containers cases are quite different and  it needs to be resolved in the guest itself. It'd be better to be addressed in another PR.

Currently we'd like to use EROFS as vm rootfs for our production. And we think our attempt on this is useful to the whole kata communities for referenece ;-)

Reply:
Yes, it's a question, it is definitely not part of this PR.